app/scripts/controllers/overview.js

```diff
@@ -498,6 +538,173 @@ angular.module('openshiftConsole')
 
         $scope.$on('$destroy', function(){
           DataService.unwatchAll(watches);
+          $(window).off('click.topologyItem');
+          $('.kube-topology g').popover('destroy');
         });
+
+        // Topology view specific code
+        var updateTimeout = null;
+
+        function updateTopology() {
+          updateTimeout = null;
+
+          var topologyRelations = [];
+          var topologyItems = { };
+
+          // Because metadata.uid is not unique among resources
+          function makeId(resource) {
+            return resource.kind + resource.metadata.uid;
+          }
+
+          // Add the services
+          angular.forEach($scope.services, function(service) {
+            topologyItems[makeId(service)] = service;
+          });
+
+          var isRecentDeployment = $filter('isRecentDeployment');
+          var isVisibleDeployment = function(deployment) {
```

Comment:
Same as above, `isVisibleDeployment` could probably be declared outside of `updateTopology`